crates/extrinsics/src/lib.rs

```diff
@@ -378,6 +384,42 @@ pub async fn fetch_wasm_code(hash: CodeHash, client: &Client) -> Result<Option<V
     Ok(pristine_bytes)
 }
 
+/// Fetch all contracts addresses from the storage using the provided client and count of
+/// requested elements starting from an optional address
+pub async fn fetch_all_contracts(
+    client: &Client,
+    count: u32,
+    from: Option<&AccountId32>,
+) -> Result<Vec<AccountId32>> {
+    let key = api::storage()
+        .contracts()
+        .contract_info_of_root()
+        .to_root_bytes();
+    let start_key = from
+        .map(|e| [key.clone(), hashing::twox_64(&e.0).to_vec(), e.0.to_vec()].concat());
+    let keys = client
+        .storage()
+        .at_latest()
+        .await?
+        .fetch_keys(key.as_ref(), count, start_key.as_deref())
+        .await?;
+
+    // StorageKey is a concatention of contract_info_of root key and
+    // Twox64Concat(AccountId)
+    let contracts = keys
+        .into_iter()
+        .map(|e| {
+            let mut account =
+                e.0.get(key.len() + 8..)
+                    .ok_or(anyhow!("Unexpected storage key size"))?;
+            AccountId32::decode(&mut account)
+                .map_err(|err| anyhow!("AccountId deserialization error: {}", err))
```

Comment:
I would extract it into a separate function for better readability

Reply:
Done